src/lib/io/sol-iio.c

```diff
@@ -963,9 +1058,27 @@ sol_iio_close(struct sol_iio_device *device)
 {
     int i;
     struct sol_iio_channel *channel;
+    struct sol_list *it = NULL, *it_next = NULL;
+    struct iio_opened_device *entry = NULL;
+    struct reader_cb_data *reader_cb_data_elem = NULL;
 
     SOL_NULL_CHECK(device);
 
+    SOL_LIST_FOREACH_SAFE (&iio_open_list, it, it_next) {
+        entry = SOL_LIST_GET_CONTAINER(it, struct iio_opened_device, list);
+        if (device->device_id == entry->device->device_id) {
+            entry->refcount--;
+            if (entry->refcount == 0) {
+                SOL_DBG("Close device%d", device->device_id);
+                sol_list_remove(it);
+                free(entry);
+            } else {
+                SOL_DBG("Cannot close device%d refcount=%d > 0", device->device_id, entry->refcount);
```

Comment:
I still think that 'Cannot' is too negative, appearing to be a problem. Maybe 'Won't'?